### PR TITLE
output/json: Set and flush thresholds

### LINKED ISSUE #1052: JSON output format doesn't include thresholds
Version
```
> k6 version                                                                                     
k6 v0.24.0
```
Test
```javascript
import { Rate } from "k6/metrics";

const rate = new Rate("some-rate");

export const options = {
    iterations: 1,
    thresholds: {
        "some-rate": ["rate>0.9"]
    }
};

export default function () {
    rate.add(true);
};
```
Run
```
 k6 run --out json=result.json  tmp.js

          /\      |‾‾|  /‾‾/  /‾/   
     /\  /  \     |  |_/  /  / /    
    /  \/    \    |      |  /  ‾‾\  
   /          \   |  |‾\  \ | (_) | 
  / __________ \  |__|  \__\ \___/ .io

  execution: local
     output: json=result.json
     script: tmp.js

    duration: -,  iterations: 1
         vus: 1, max: 1

    done [==========================================================] 1 / 1

    data_received........: 0 B     0 B/s
    data_sent............: 0 B     0 B/s
    iteration_duration...: avg=61.01µs min=61.01µs med=61.01µs max=61.01µs p(90)=61.01µs p(95)=61.01µs
    iterations...........: 1       0/s
  ✓ some-rate............: 100.00% ✓ 1   ✗ 0  
    vus..................: 1       min=1 max=1
    vus_max..............: 1       min=1 max=1
```
Json
```json
{"type":"Metric","data":{"name":"some-rate","type":"rate","contains":"default","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"some-rate"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.233005026+03:00","value":1,"tags":{"group":""}},"metric":"some-rate"}
{"type":"Metric","data":{"name":"data_sent","type":"counter","contains":"data","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"data_sent"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.233007615+03:00","value":0,"tags":{"group":""}},"metric":"data_sent"}
{"type":"Metric","data":{"name":"data_received","type":"counter","contains":"data","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"data_received"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.233007615+03:00","value":0,"tags":{"group":""}},"metric":"data_received"}
{"type":"Metric","data":{"name":"iteration_duration","type":"trend","contains":"time","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"iteration_duration"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.233007615+03:00","value":0.061017,"tags":{"group":""}},"metric":"iteration_duration"}
{"type":"Metric","data":{"name":"iterations","type":"counter","contains":"default","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"iterations"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.233012443+03:00","value":1,"tags":null},"metric":"iterations"}
{"type":"Metric","data":{"name":"vus","type":"gauge","contains":"default","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"vus"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.269144407+03:00","value":1,"tags":null},"metric":"vus"}
{"type":"Metric","data":{"name":"vus_max","type":"gauge","contains":"default","tainted":null,"thresholds":[],"submetrics":null,"sub":{"name":"","parent":"","suffix":"","tags":null}},"metric":"vus_max"}
{"type":"Point","data":{"time":"2019-06-14T17:42:54.269144407+03:00","value":1,"tags":null},"metric":"vus_max"}
```

Let's look at our `some-metric` line from prev file:
```json
{
  "type": "Metric",
  "data": {
    "name": "some-rate",
    "type": "rate",
    "contains": "default",
    "tainted": null,
    "thresholds": [], // <-- ¯\_(ツ)_/¯ 
    "submetrics": null,
    "sub": {
      "name": "",
      "parent": "",
      "suffix": "",
      "tags": null
    }
  },
  "metric": "some-rate"
}
```
So, as you can see, there is no threshold name in the "thresholds" array, as it described in the [spec](https://docs.k6.io/docs/results-output#section-json-output). I can't know that some thresholds was defined. It's really big problem when I need to analyze results, because I don't know were all thresholds passed or not.

### PR DESCRIPTION
**Pull request by codebien**
Added the ability to set and flush the thresholds implementing the WithThresholds interface as required from the Engine.

The SetThresholds implementation is the same as the cloud version. If you like the idea, I could also refactor out it to centralize the logic.

Closes #1052

### CODE REVIEW

File: output/json/json.go [hunk 1]
```diff
@@ -32,7 +32,6 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// TODO: add option for emitting proper JSON files (https://github.com/loadimpact/k6/issues/1052)
```
Comment (na--) [inline output/json/json.go:35]: Ah, this was a wrong copy-paste on my part, the actual issue in the brackets here should have been https://github.com/loadimpact/k6/issues/737, which is still not fixed. Can you please add the `TODO` back, but with the correct link this time? :sweat_smile:

**Review by na-- — COMMENTED** [review]
Thanks, this LGTM! 

Can you please rebase the PR on top of the latest `master`, the new CI check we added on Friday (https://github.com/loadimpact/k6/pull/1885) is failing the build, because GitHub runs the CI on top an imaginary merge commit between your branch and master, but the CI check uses your actual PR commits (because it essentially does `go get`).

**Comment by codebien** [comment]
>Can you please rebase the PR on top of the latest master
@na-- I rebased, but when I opened the PR I was already based on the latest version. I forked on Saturday.

Can the check work with a fork? Because from my understanding it's trying to exec a `go get` using my latest commit from my forked version using the `k6` path.

**Comment by na--** [comment]
Ah, yeah, sorry... :disappointed: I created https://github.com/loadimpact/k6/issues/1891 and we'll fix it soon after we release v0.31.0, which would hopefully happen tomorrow. The `xk6` CI checks are not blocking the PR merge, so for now please just fix the comment I mentioned inline and we can merge the PR without you rebasing again after we release v0.31.0.

**Comment by codebien** [comment]
>so for now please just fix the comment I mentioned

Sorry, I forgot to mention that the fix is already included in the RS and pushed.

File: output/json/json.go (hunk 1 above)
Comment (codebien) [reply output/json/json.go:35]: @na-- Done

**Review by na-- — APPROVED** [review]

**Comment by na--** [comment]
@codebien, sorry to bother you again, but could you rebase your PR on top of k6's `master` once again? @imiric should have fixed the xk6 CI bug with https://github.com/loadimpact/k6/pull/1894.

**Comment by codebien** [comment]
@na-- Done

**Review by na-- — APPROVED** [review]
Thanks! Seems like the check is now ok :tada:

**Review by imiric — APPROVED** [review]
LGTM, thanks for your contribution!

**Review by MStoykov — APPROVED** [review]

**Pull request merged**